lib/URI.pm

```diff
@@ -86,10 +92,35 @@ sub _init
 }
 
 
+#-- Version: 5.11+
+#   Since the complete URI will be percent-encoded including '[' and ']',
+#   we selectively unescape square brackets from the autority/host part of the URI.
+#   Derived modules that implement _uric_escape() should take this into account
+#   if they do not rely on URI::_uric_escape().
+#   No unescaping is performed for the userinfo@ part of the authority part.
+sub _fix_uric_escape_for_host_part {
+  return if HAS_RESERVED_SQUARE_BRACKETS;
+  return if $_[0] !~ /%/;
+
+  if ($_[0] =~ m,^((?:$URI::scheme_re:)?)//([^/?\#]*)(.*)$,os) {
+    my $orig          = $2;
+    my ($user, $host) = $orig =~ /^(.*@)?([^@]*)$/;
+    $user  ||= '';
+    my $port = $host =~ s/(:\d+)$// ? $1 : '';
+    #TODO: die() here if scheme indicates TCP/UDP and port is out of range [0..65535] ?
+    #TODO: count substitutions and die() here if count('[') != count(']') > 1 ?  (imbalanced or too many)
```

Comment:
Are these for later, or did you forget to do that part?

Reply:
Just a  note to myself or perhaps and indicator to the location where a patch could be applied.
Esp. with line 110, I didn't want to interfere with the discussion of issue #70. Theses should be different patches.
Sometimes, it requires more than a regular expression to check proper syntax.
This rises the general question, if URI.pm shall be more strict or more relaxed with respect to syntax checks.
My concern is to break legacy code being too strict.
Some ideas:  `use URI  ( strict => 'all' ); # and factor out checks into URI::strict`.
This concern also drove my decision to restore the previous behavior using an ENVironment variable.
Usage of '[' and  ']' once was legal before introduction of IPv6 address syntax.
 
#TODO:  is a tag that my tools check for. What is your suggestion? Rename the tag?  #HINT: ?

Reply:
I'll remove line 111 and change #TODO:  into #MAINT:.